### DISCUSSION TITLE
Support for chain-specific transactions (eg: CIP42)

### DISCUSSION THREAD

**Discussion by nicolasbrugneaux** [wevm/wagmi#2817] [Ideas]
Hello there 🌱

Following the work of @aaronmgdr in the following PRs:
- wagmi-dev/viem/pull/691
- wagmi-dev/viem/pull/755

We're now able to do something this:
```ts
const sendTransaction = (tx: SendTransactionParameters<typeof celo>) => {
  return localAccountClient.sendTransaction(tx)
}

const hash = await sendTransaction({
  feeCurrency: cUSDAddress,
  value: BigInt(...),
  to: '0x...',
})
```

Recently, I've been trying to use it via wagmi.

So my initial guess was to do something along these lines:
```ts
  const { config } = usePrepareSendTransaction<typeof celo>({
    feeCurrency: cUSDAddress,
    value: ...,
    to: '0x...',
  });
```

But it seems that feeCurrency just gets dropped. (and obviously `usePrepareSendTransaction` isn't a generic, but this was my naive attempt)

After some digging, I've hit a wall as this file `./packages/core/src/actions/transactions/sendTransaction.ts` doesn't seem to be able to handle extraneous args (in our case: `feeCurrency`). 

I've been trying to come up with a neat PR that would solve the issue, but this is – after all – a fairly foreign codebase, and I'd really appreciate some pointers if you have some ideas how to solve this :)

**Comment by tmm** [comment]
This is something we are currently working on and will get rolled out in wagmi v2.

**Reply by nicolasbrugneaux** [reply]
Amazing news. Looking forward to it. Let us know if you need some beta testing too 🙏

**Reply by arthurgousset** [reply]
+1 that's great news 💯 To help plan on our side, what order of magnitude can I use to think about the wagmi v2 release @tmm? Is it days, weeks, months away (roughly speaking)?

**Reply by nicolasbrugneaux** [reply]
Hey there @tmm, where can I follow the v2 progress (if anywhere)? Thank you in advance! 🙏

**Reply by tmm** [reply]
We'll be announcing v2 alpha shortly. You can follow along on the alpha branch.

**Reply by nicolasbrugneaux** [reply]
Cool thanks! 🥳
